Refactor: remove minute visible name from Connection dialogue icons

SlySven opened the pull request:
We have been trying to hide the text associated with the `QListWidgetItems` in the connection dialogue by setting the font size to the minimum of `1` and by setting it's colour to be white. This is not effective when the background is not white - which is likely for a "Dark" desktop environment. The only way to successfully hide the text is, I think, to not have any!

However the text was being used programmatically, so the best way to use a `QListWidget` in this manner is to store the text elsewhere in each item's structure. Fortunately Qt provides for this with the user data functionality which allows multiple data items (based on the `QVariant` class) to be stored within each `QListWidgetItem` using an integer key to denote the type of the data. Ironically the text, icon and other details for each item are ALSO stored in this way - however for non-Qt internal use the lowest integer key that is to be used is `Qt::UserRole` - which I have assigned to the `(const int) dlgConnectionProfiles::csmNameRole` static value. It is quite possible that a redesign of the Connect Profiles dialogue may use this system to store/cache more details about each profile in the future!

The only issue with this is the lack of a: `QListWidget::findData(const QVarient&data, int role = Qt::UserRole ...)`method (c.f. `QComboBox::findData(...)`) so I have had to provide a: `(QList<QListWidgetItem*>) findData(const QListWidget&, const QVariant&,                         const int role = Qt::UserRole) const` method to fill in this gap and to replace the previous `QListWidget::find(...)` that examined each item's text.

Also removed local `(QString) profile` from: `(void) dlgConnectionProfiles::slot_item_clicked(QListWidgetItem*)` as it is merely a redundant duplicate of another local `(QString) profile_name`.

Renamed local `(QListWidgetItem*) pM` from: `(void) dlgConnectionProfiles::fillout_form()` to: `(QListWidgetItem*) pItem` as that reflects the name used throughout the rest of the class for this type of variable.

Changed a `(QLabel*)::setText(tr(""))` call to the more explicit and less stupid `(QLabel*)::clear()`. Similarly changed a `(QWidget*)::setToolTip("")` to a `(QWidget*)::setToolTip(QString())`.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Refactor: remove minute visible name from Connection dialogue icons

We have been trying to hide the text associated with the `QListWidgetItems`
in the connection dialogue by setting the font size to the minimum of `1`
and by setting it's colour to be white. This is not effective when the
background is not white - which is likely for a "Dark" desktop environment.
The only way to successfully hide the text is, I think, to not have any!

However the text was being used programmatically, so the best way to use
a `QListWidget` in this mannar is to store the text elsewhere in each
item's structure. Fortunately Qt provides for this with the user data
functionality which allows multiple data items (based on the `QVariant`
class) to be stored within each `QListWidgetItem` using an integer key to
denote the type of the data. Ironically the text, icon and other details
for each item are ALSO stored in this way - however for non-Qt internal
use the lowest integer key that is to be used is `Qt::UserRole` - which
I have assigned to the `(const int) dlgConnectionProfiles::csmNameRole`
static value. It is quite possible that a redesign of the Connect Profiles
dialogue may use this system to store/cache more details about each profile
in the future!

The only issue with this is the lack of a:
`QListWidget::findData(const QVarient&data, int role = Qt::UserRole ...)`
method (c.f. `QComboBox::findData(...)`) so I have had to provide a:
`(QList<QListWidgetItem*>) findData(const QListWidget&, const QVariant&,
                                      const int role = Qt::UserRole) const;
method to fill in this gap and to replace the previous
`QListWidget::find(...)` that examined each item's text.

Also removed local `(QString) profile` from:
`(void) dlgConnectionProfiles::slot_item_clicked(QListWidgetItem*)` as it
is merely a redundent duplicate of another local `(QString) profile_name`.

Renamed local `(QListWidgetItem*) pM` from:
`(void) dlgConnectionProfiles::fillout_form()` to:
`(QListWidgetItem*) pItem` as that reflects the name used throughout the
rest of the class for this type of variable.

Changed a `(QLabel*)::setText(tr(""))` call to the more explicit and less
stupid `(QLabel*)::clear()`. Similarly changed a
`(QWidget*)::setToolTip("")` to a `(QWidget*)::setToolTip(QString())`.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented on src/dlgConnectionProfiles.cpp:
```diff
@@ -492,9 +482,13 @@ void dlgConnectionProfiles::slot_save_name()
-        auto newProfileIcon = profiles_tree_widget->findItems(newProfileName, Qt::MatchExactly).first();
-        profiles_tree_widget->setCurrentItem(newProfileIcon);
-        slot_item_clicked(newProfileIcon);
+        // However there is not a findData(...) method for QListWidgets...
```
I don't think we need this comment, it doesn't add much and the less things we have to read - the better.

SlySven replied:
I ~~was thinking of raising~~ *have raised* it upstream as a desirable enhancement: https://bugreports.qt.io/browse/QTBUG-85402 ...

... but I guess I can cut the comment out.

vadi2 reviewed (approved):
Great improvement, thank you!

SlySven commented on src/dlgConnectionProfiles.cpp:
```diff
@@ -492,9 +482,13 @@ void dlgConnectionProfiles::slot_save_name()
-        auto newProfileIcon = profiles_tree_widget->findItems(newProfileName, Qt::MatchExactly).first();
-        profiles_tree_widget->setCurrentItem(newProfileIcon);
-        slot_item_clicked(newProfileIcon);
+        // However there is not a findData(...) method for QListWidgets...
```
```suggestion
```

SlySven committed:
Update src/dlgConnectionProfiles.cpp

Remove unneeded command as per peer-review suggestion.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented on src/dlgConnectionProfiles.h:
```diff
@@ -44,6 +44,9 @@ class dlgConnectionProfiles : public QDialog, public Ui::connection_profiles
     void accept() override;
+    QList<QListWidgetItem*> findData(const QListWidget& listWidget, const QVariant& what, const int role = Qt::UserRole) const;
+
+    const static int csmNameRole{Qt::UserRole};
```
```suggestion
    static const int csmNameRole{Qt::UserRole};
```

SlySven replied:
CodeFactor wants the `static` at the beginning of the type specification... :roll_eyes:

SlySven committed:
Revise: src/dlgConnectionProfiles.h

CodeFactor says that the `static` keyword should come first (before the `const` one).

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 reviewed (approved):
Consistency is good!
